### PR TITLE
refactor: consolidate player driver

### PR DESCRIPTION
**Pull request by tenthirtyam**
### Description

Consolidates `Player5Driver` and `Player6Driver` to `PlayerDriver` within `driver_player.go`.

### Testing

- [x] Basic
- [ ] End-to-End

```console
packer-plugin-vmware on  refactor/consolidate-player-driver
➜ go fmt ./...

packer-plugin-vmware on  refactor/consolidate-player-driver
➜ make generate                                            
2024/07/18 18:07:15 Copying "docs" to ".docs/"
2024/07/18 18:07:15 Replacing @include '...' calls in .docs/
Compiling MDX docs in '.docs' to Markdown in '.web-docs'...

packer-plugin-vmware on  refactor/consolidate-player-driver
➜ make build   

packer-plugin-vmware on  refactor/consolidate-player-driver
➜ make test    
?       github.com/hashicorp/packer-plugin-vmware       [no test files]
?       github.com/hashicorp/packer-plugin-vmware/version       [no test files]
ok      github.com/hashicorp/packer-plugin-vmware/builder/vmware/common 6.704s
ok      github.com/hashicorp/packer-plugin-vmware/builder/vmware/iso    2.066s
ok      github.com/hashicorp/packer-plugin-vmware/builder/vmware/vmx    2.544s

packer-plugin-vmware on  refactor/consolidate-player-driver 
➜ make dev     
packer plugins install --path packer-plugin-vmware "github.com/hashicorp/vmware"
Successfully installed plugin github.com/hashicorp/vmware from /Users/ryan/Library/Mobile Documents/com~apple~CloudDocs/Code/Personal/packer-plugin-vmware/packer-plugin-vmware to /Users/ryan/.packer.d/plugins/github.com/hashicorp/vmware/packer-plugin-vmware_v1.0.12-dev_x5.0_darwin_arm64
```

### CODE REVIEW

**Comment by tenthirtyam** [comment]
Ready for initial review; e2e validations are still in flight with the latest updates.

**Review by lbajolet-hashicorp — APPROVED** [review]
Left a couple nits, but overall this LGTM, thanks @tenthirtyam!

Pre-approving now, we can merge once you've had a chance to address my comments

File: builder/vmware/common/driver_player.go [hunk 1]
```diff
@@ -132,32 +137,43 @@ func (d *Player5Driver) Start(vmxPath string, headless bool) error {
...
+		}
+
 		return err
 	}
 
 	return nil
 }
 
-func (d *Player5Driver) SuppressMessages(vmxPath string) error {
+func (d *PlayerDriver) SuppressMessages(vmxPath string) error {
```
[the remark this answers is not in the document]
Comment (tenthirtyam) [reply builder/vmware/common/driver_player.go:155]: Yep, there's a SupressMessages for each of the hypervisor choices as an interface in the driver.

File: builder/vmware/common/driver_player.go [hunk 2]
```diff
@@ -132,32 +137,43 @@ func (d *Player5Driver) Start(vmxPath string, headless bool) error {
...
 	return nil
 }
 
-func (d *Player5Driver) Verify() error {
+func (d *PlayerDriver) Verify() error {
 	var err error
+
+	log.Printf("Checking %s paths...", playerProductName)
```
[the remark this answers is not in the document]
Comment (tenthirtyam) [reply builder/vmware/common/driver_player.go:162]: Good idea! I'll update this item.

[the remark this answers is not in the document]
Comment (tenthirtyam) [reply builder/vmware/common/driver_player.go]: Updated in 5b336cb.

[the remark this answers is not in the document]
Comment (lbajolet-hashicorp) [reply builder/vmware/common/driver_player.go]: I would suggest also updating the other `log.Printf` added with this PR with the same kind of idea, let me know if you're OK with the idea, we can proceed with a merge regardless as it's a bit of a nit, but I like consistency :)

[the remark this answers is not in the document]
Comment (tenthirtyam) [reply builder/vmware/common/driver_player.go]: Will do!

[the remark this answers is not in the document]
Comment (lbajolet-hashicorp) [reply builder/vmware/common/driver_player.go]: Just to update: I see the logs have not been updated, did you still want to proceed with the change, or should we merge this in the current state?

[the remark this answers is not in the document]
Comment (tenthirtyam) [reply builder/vmware/common/driver_player.go]: doh! i'll look at it over the weekend for you!

[the remark this answers is not in the document]
Comment (tenthirtyam) [reply builder/vmware/common/driver_player.go]: Take a look at https://github.com/hashicorp/packer-plugin-vmware/compare/5b336cb147310482d5ebb1bf32c8fe768f33477f..7229c554856793aae1ef66357d6fb19f1b72914b and https://github.com/hashicorp/packer-plugin-vmware/compare/7229c554856793aae1ef66357d6fb19f1b72914b..ca282efb57408ead3599b89e0a18c357d55ca993 and let me know what you think.

**Review by lbajolet-hashicorp — APPROVED** [review]
LGTM! Thanks for the rerolls @tenthirtyam

Merging now!

**Pull request merged**